### PR TITLE
Feature.collection to s 195

### LINKED ISSUE #195: Collection name in class names should be changed to 's'
This is to make this repo and the modules within a little easier to read to the user. When instantiating a pool, the path was once:

bigip.sys.poolcollection.pool

Not it will be:

bigip.sys.pools.pool

### PR DESCRIPTION
**Pull request by zancas**
@pjbreaux Ask...   and ya shall receive.

Fixes #195, please see commit message for details.

### CODE REVIEW

File: f5/bigip/ltm/__init__.py [hunk 1]
```diff
@@ -13,12 +13,12 @@
...
 #
 from f5.bigip.ltm.monitor import Monitor
-from f5.bigip.ltm.nat import NATCollection
-from f5.bigip.ltm.policy import PolicyCollection
-from f5.bigip.ltm.pool import PoolCollection
-from f5.bigip.ltm.rule import RuleCollection
-from f5.bigip.ltm.snat import SNATCollection
-from f5.bigip.ltm.virtual import VirtualCollection
+from f5.bigip.ltm.nat import NATs
+from f5.bigip.ltm.policy import Policys
```
Comment (pjbreaux) [inline f5/bigip/ltm/__init__.py]: As long as everyone is fine with Policys, I am fine. Just bringing attention to it.

File: f5/bigip/ltm/monitor.py [hunk 2]
```diff
@@ -186,75 +186,75 @@ def __init__(self, monitor):
...
+        super(LDAPs, self).__init__(monitor)
         self._meta_data['allowed_lazy_attributes'] = [LDAP]
         self._meta_data['attribute_registry'] =\
             {'tm:ltm:monitor:ldap:ldapstate': LDAP}
 
 
 class LDAP(UpdateMonitorMixin, Resource):
-    def __init__(self, ldap_collection):
-        super(LDAP, self).__init__(ldap_collection)
+    def __init__(self, ldap_s):
```
Comment (pjbreaux) [inline f5/bigip/ltm/monitor.py]: Some of these types of parameter names could be 'ldaps' without the underscore.

File: f5/bigip/ltm/monitor.py [hunk 3]
```diff
@@ -552,45 +552,45 @@ def __init__(self, monitor):
...
+        super(TCP, self).__init__(tcp_s)
         self._meta_data['required_json_kind'] =\
             'tm:ltm:monitor:tcp:tcpstate'
 
 
-class TCP_EchoCollection(Collection):
+class TCP_Echos(Collection):
     def __init__(self, monitor):
-        super(TCP_EchoCollection, self).__init__(monitor)
+        super(TCP_Echos, self).__init__(monitor)
```
Comment (pjbreaux) [inline f5/bigip/ltm/monitor.py]: This could be slightly confusing. Folks might natually say TCP_Echoes

File: f5/bigip/mixins.py [hunk 4]
```diff
@@ -83,13 +83,13 @@ def __getattr__(self, name):
         # but many nat attributes just like the BIGIP device.
         for lazy_attribute in self._meta_data['allowed_lazy_attributes']:
             if name == lazy_attribute.__name__.lower():
-                iface_collection = lazy_attribute(self)
+                iface_s = lazy_attribute(self)
```
Comment (pjbreaux) [inline f5/bigip/mixins.py]: iface_s could be ifaces

File: f5/bigip/net/l2gre.py [hunk 5]
```diff
@@ -14,10 +14,10 @@
 #
 
 from f5.bigip import exceptions
-from f5.bigip.rest_collection import icontrol_rest_folder
-from f5.bigip.rest_collection import log
-from f5.bigip.rest_collection import prefixed
-from f5.bigip.rest_collection import strip_folder_and_prefix
+from f5.bigip.rest_s import icontrol_rest_folder
```
Comment (pjbreaux) [inline f5/bigip/net/l2gre.py]: Module doesn't exist anymore -@zancas

File: f5/bigip/net/route.py [hunk 6]
```diff
@@ -14,8 +14,8 @@
 #
 
 from f5.bigip import exceptions
-from f5.bigip.rest_collection import icontrol_rest_folder
-from f5.bigip.rest_collection import log
+from f5.bigip.rest_s import icontrol_rest_folder
```
Comment (pjbreaux) [inline f5/bigip/net/route.py]: Module doesn't exist anymore.

File: f5/bigip/net/vxlan.py [hunk 7]
```diff
@@ -17,10 +17,10 @@
 
 from eventlet import greenthread
 from f5.bigip import exceptions
-from f5.bigip.rest_collection import icontrol_rest_folder
-from f5.bigip.rest_collection import log
-from f5.bigip.rest_collection import prefixed
-from f5.bigip.rest_collection import strip_folder_and_prefix
+from f5.bigip.rest_s import icontrol_rest_folder
```
Comment (pjbreaux) [inline f5/bigip/net/vxlan.py]: Module no longer exists.

File: f5/bigip/resource.py [hunk 8]
```diff
@@ -406,8 +409,8 @@ def _update(self, **kwargs):
         temp_meta = self.__dict__.pop('_meta_data')
 
         # Need to remove any of the Collection objects from self.__dict__
-        # because these are sub-collections and _meta_data and
-        # other non-BIGIP attrs are not removed from the sub-collections
+        # because these are sub-ss and _meta_data and
```
Comment (pjbreaux) [inline f5/bigip/resource.py]: Hmm this docstring is a little weird. Reword as best we can.

File: f5/bigip/resource.py [hunk 9]
```diff
@@ -406,8 +409,8 @@ def _update(self, **kwargs):
...
+        # other non-BIGIP attrs are not removed from the sub-ss
```
Comment (pjbreaux) [inline f5/bigip/resource.py]: Same as above.

File: f5/bigip/sys/system.py [hunk 10]
```diff
@@ -16,7 +16,7 @@
 # pylint: disable=broad-except
 
 from f5.bigip import exceptions
-from f5.bigip.rest_collection import log
+from f5.bigip.rest_s import log
```
Comment (pjbreaux) [inline f5/bigip/sys/system.py]: Module no longer exist.

File: f5/bigip/sys/test/test_sys_application.py [hunk 11]
```diff
@@ -29,7 +29,7 @@
 
 
 KIND_MISMATCH = {
-    "kind": "tm:sys:application:service:servicecollectionstate",
+    "kind": "tm:sys:application:service:servicesstate",
```
Comment (pjbreaux) [inline f5/bigip/sys/test/test_sys_application.py]: Kind is wrong here.

**Comment by pjbreaux** [comment]
I've checked it over. Punting to @swormke, so he can take a look and see if it suits him well enough.

File: f5/bigip/ltm/monitor.py (hunk 3 above)
Comment (zancas) [inline f5/bigip/ltm/monitor.py]: For now I think it's better to leave it in than add special edge-case logic.

**Comment by zancas** [comment]
@pjbreaux   I think I've addressed all your comments.

**Comment by pjbreaux** [comment]
Merging. Thanks for updating the local vars.

**Pull request merged**